nipst/nipst.go

```diff
@@ -217,9 +217,13 @@ func (nb *NIPSTBuilder) BuildNIPST(challenge *common.Hash) (*types.NIPST, error)
 
 func (nb *NIPSTBuilder) IsPostInitialized() bool {
 	dir := shared.GetInitDir(nb.postCfg.DataDir, nb.id)
-	_, err := os.Stat(dir)
-	if os.IsNotExist(err) {
-		nb.log.Info("could not find init files at %v", dir)
+	infos, err := ioutil.ReadDir(dir)
+	if err != nil {
+		nb.log.WithFields(log.Err(err), log.String("dir", dir)).Error("failed to look for init files")
+		return false
+	}
+	if len(infos) < 2 {
+		nb.log.WithFields(log.Int("num_files", len(infos))).Info("could not find init files")
 		return false
 	}
 	return true
```

Comment:
Having more than 1 files isn't reliable metric. First, there might be other os-related files there. Second, the post might be in a non-complete state (in-progress, paused, crashed).

Best way to implement this is to inspect the initialization state. Support for this was added here: https://github.com/spacemeshos/post/pull/22.

Reply:
@zalmen told me you have a more reliable implementation and that I should use this temp hack in the interim.